addons/loyalty/models/loyalty_reward.py

```diff
@@ -148,7 +148,10 @@ def _compute_all_discount_product_ids(self):
     @api.depends('reward_product_id', 'reward_product_tag_id', 'reward_type')
     def _compute_multi_product(self):
         for reward in self:
-            products = reward.reward_product_id + reward.reward_product_tag_id.product_ids
+            products = list(set(
+                [reward.reward_product_id.id] if reward.reward_product_id else []
+                + reward.reward_product_tag_id.product_ids.ids
+            ))
```

Comment:
```suggestion
            products = reward.reward_product_id | reward.reward_product_tag_id.product_ids
```
This should work too to avoid having duplicates in `products` (and avoid the list set conversion, which can be avoided as recordsets support [set operations](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#operations))